src/Silverback.Integration/Messaging/Configuration/EndpointsConfigurationBuilderAddOutboundExtensions.cs

```diff
@@ -183,22 +220,44 @@ public static class EndpointsConfigurationBuilderAddOutboundExtensions
         ///     Specifies whether the producers must be immediately instantiated and connected. When <c>false</c> the
         ///     <see cref="IProducer" /> will be created only when the first message is about to be produced.
         /// </param>
+        /// <param name="idempotentRegistration">
+        ///     Only adds the endpoint when it is not already registered.
+        ///     <remarks>
+        ///         Validates only against static routes based on the message type (see <see cref="StaticOutboundRouter"/>),
+        ///         not routes that are calculated at runtime based on the message.
+        ///     </remarks>
+        /// </param>
         /// <returns>
         ///     The <see cref="IEndpointsConfigurationBuilder" /> so that additional calls can be chained.
         /// </returns>
         public static IEndpointsConfigurationBuilder AddOutbound(
             this IEndpointsConfigurationBuilder endpointsConfigurationBuilder,
             Type messageType,
             IEnumerable<IProducerEndpoint> endpoints,
-            bool preloadProducers = true)
+            bool preloadProducers = true,
+            bool idempotentRegistration = false)
```

Comment:
The more I think about it and the more I'm convinced that this should be a general setting at an higher level. It doesn't make a whole lot of sense to be idempotent for a registration and not another and leave more room for configuration mistakes.